### DISCUSSION TITLE
Pipeline / CI : Pages sporadically fail to load via cy.visit()

### DISCUSSION THREAD

**Discussion by robrich7** [cypress-io/cypress#20651] [Questions and Help]
Hello all, 

we have the problem that tests often fail because pages do not load. This problem only occurs when running Cypress via pipeline, in our case the Bitbucket pipeline. Cypress version is `9.5.1` and browser is `chrome` or `electron`, it doesnt matter.
Locally all tests always run cleanly.

I have defined a `BaseUrl` in the `cypress.json` and change via `cy.visit()` only the endpoints.
So as an example: `cy.visit('/login')  `
Here is an example of what it looks like in the dashboard video.

![image](https://user-images.githubusercontent.com/70799178/158663906-15fd8ddf-a7e3-46dd-9bed-17fca516cd35.png)

As I said, if you start the tests locally via GUI or console, then there are no problems. The pages are always loaded but on each run via pipeline several tests are aborted because the pages are not loaded. Also the retries fail. 
I don't understand what could be the reason for this. 

Could it be because there are multiple `cy.visit()` functions during the execution of a test? So I often use different visit() commands in `beforeEach` and `afterEach`. Does Cypress have trouble with this? But then it should have the same problems locally. 

Do you have any idea?

**Comment by jjhelguero** [comment]
It might be something with the CI set up. Have any changes been introduced before this started happening?

**Reply by robrich7** [reply]
we have introduced a dynamic baseURL via tag but I am not sure if the problem was there before. 
With using a tag, we can change the baseURL that Cypress accesses. It looks like this: 

```
        script: |
          #!/usr/bin/env bash
          set -e
          npm i cypress@9.5.1
          npm i dayjs
          if [[ $REVISION == d1* ]]; then CYPRESS_BASE_URL="staging_url" ; fi
          if [[ $REVISION == p1* ]]; then CYPRESS_BASE_URL="production_url" ; fi
          echo -n "#### Starting tests on $CYPRESS_BASE_URL"
          PARAMS="--browser chrome"
          if [[ $REVISION == *-rec-* ]]; then
            echo " with Record enabled"
            PARAMS="$PARAMS --record --key $(params.CYPRESS_RECORD_KEY) --group frontend-tests --ci-build-id $MY_POD_NAME"
          fi
          echo
          export CYPRESS_BASE_URL
          npx cypress run $PARAMS
```

but I can't imagine that this is the reason. I rather suspect that it is something in the interaction with Cypress and the Docker image or with Cypress and the pipeline.

**Reply by robrich7** [reply]
after Debugging it seems to be due to this error: 

`cypress:https-proxy received error on client browserSocket { err: Error: read ECONNRESET     at TCP.onStreamRead (node:internal/stream_base_commons:211:20) { errno: -104, code: 'ECONNRESET', syscall: 'read' }, url:....`

However, I can't find a workaround on the Internet, which means I can't verify if it's this. has anyone ever had to deal with this and can help me?
